packages/nextjs/app/configure/_components/DownloadContracts.tsx

```diff
@@ -0,0 +1,124 @@
+      provider.getClassAt(address),
+      provider.getClassHashAt(address),
+    ]);
+
+    const contractsData = {
+      [targetNetwork.network]: {
+        [symbol]: {
+          address,
+          classHash,
+          abi: apiResponse.abi,
+        },
+      },
+    };
+    const jsonString = `data:text/json;charset=utf-8,${encodeURIComponent(JSON.stringify(contractsData, null, 2))}`;
+    const link = document.createElement("a");
+    link.href = jsonString;
+    link.download = "contracts.json";
+    link.click();
+  };
+
+  return (
+    <div className="justify-center items-center flex">
+      <div className="p-5 border-gradient rounded-[5px] w-full max-w-3xl">
+        <div className="text-xl mb-4">
+          Fetch Contract Configuration File from Contract Address
+        </div>
+        <div className="mb-6">
+          <p>
+            <strong>Instructions:</strong>
+          </p>
+          <p>
+            Currently, users are limited to the `predeployedContracts.ts`
+            provided by default in the monorepo. This tool allows you to fetch
+            the ABI of a contract by entering its address and then download a
+            configuration file that can be used to replace or append to your
+            local `predeployedContracts.ts` file, enabling debugging in the
+            `/debug` page.
+          </p>
+          <ol className="list-decimal list-inside mt-2 space-y-1">
+            <li>Enter the contract address in the input field.</li>
+            <li>
+              Select the contract type (`Eth` or `Strk`) from the dropdown.
+            </li>
+            <li>Click the Download Contract File button.</li>
+            <li>
+              The tool will fetch the ABI, address, and classHash from the
+              network and generate a JSON configuration file.
+            </li>
+            <li>
+              Download the file and replace or append it to your local
+              `predeployedContracts.ts` file.
+            </li>
+            <li>
+              Use the `/debug` page to interact with and test the contract using
+              the scaffold hooks.
+            </li>
+          </ol>
+          <p className="mt-4">
+            Ensure that the format of the ABI matches the expected format in
+            your project before replacing the file.
+          </p>
+        </div>
+        {targetNetwork && (
+          <div className="my-4 flex text-md">
+            <div className="w-24 font-medium break-words text-function">
+              Network:
+            </div>
+            <span className="text-network">{targetNetwork.name}</span>
+          </div>
+        )}
+        <div className="flex items-center my-2">
+          <div className="w-24 font-medium break-words text-function">
+            Contract:
+          </div>
+          <select
+            value={symbol}
+            onChange={(e) => setSymbol(e.target.value as "Eth" | "Strk")}
```

Comment:
We shouldnt be an option between "Eth" and "Strk" only, we should allow the user to set a Name to the downloaded contract, and this name has to be different ( ignoring case ) to all the contracts on `deployedContracts` and `preDeployedContracts`.